aether-ui/aether/ui/assets/apps/pipeline/components/PipelineCard.jsx

```diff
@@ -44,12 +67,23 @@ class PipelineCard extends Component {
           <span className='pipeline-name'>
             // { pipeline.name }
           </span>
-          <div className='pipeline-actions'>
-            <PipelineOptions
-              delete={this.props.delete}
-            />
-            <ContractAddButton pipeline={pipeline} history={this.props.history} />
-          </div>
+          <PipelineRename
```

Comment:
The logic here is a little weird: 
- if `state.renameView === 'form'`, `PipelineRename` will render a form, and the `pipeline-actions` div will not be rendered
- else `PipelineRename` will render `null`, and the `pipeline-actions` div _will_ be rendered
- also, `renameView` has two possible values, which actually makes it a good candidate to simplify into a boolean, e.g. `isRenaming`

It would be nice to make this a bit less spaghetti-ish, so that this render method looks more like:

```
{ this.state.isRenaming ?
  <PipelineRename /> :
  <PipelineActions /> }
```

That way, `PipelineRename` doesn't need the `view` prop.

Reply:
(Where `PipelineOptions` could be a very simple little functional component that just needs a couple of functions passed in as props)

Reply:
Yeah. the hide on `pipeline-actions` was from the 1st implementation, now since a modal is been used, hiding the actions is no longer necessary. Refactoring

Reply:
Thanks @lordmallam , this is better - but do you think it would be possible to implement it like I outlined above, so that this is moved out to a simple functional component:
```
          <div className='pipeline-actions'>
            <PipelineOptions
              delete={this.props.delete}
              rename={this.onRename}
            />
            <ContractAddButton pipeline={pipeline} history={this.props.history} />
          </div>
```
called, for example, `PipelineActions`. Then the `PipelineCard` render function can become:
```
{ this.state.isRenaming ?
  <PipelineRename /> :
  <PipelineActions /> }
```
which is a lot more readable. This means you can also remove the `show` prop from `PipelineRename`, which will mean that you can easily get rid of the `renderForm` method.